ui-speedspacechart/src/components/layers/SpeedLimitTagsLayer.tsx

```diff
@@ -1,21 +1,39 @@
-import React, { useEffect, useRef } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 
 import type { Store, tooltipInfos } from '../../types/chartTypes';
 import Tooltip from '../common/Tooltip';
 import { LINEAR_LAYERS_HEIGHTS } from '../const';
-import { drawSpeedLimitTags, computeTooltip } from '../helpers/drawElements/speedLimitTags';
+import {
+  drawSpeedLimitTags,
+  computeTooltip,
+  loadSpeedLimitTagsImages,
+} from '../helpers/drawElements/speedLimitTags';
 
 type SpeedLimitTagsLayerProps = {
   width: number;
   marginTop: number;
   store: Store;
 };
+type ImagesType = {
+  questionImage: HTMLImageElement | null;
+  alertFillImage: HTMLImageElement | null;
+};
 const TOOLTIP_HEIGHT = 40;
 const MARGIN_ADJUSTMENT = 2;
 
 const SpeedLimitTagsLayer = ({ width, marginTop, store }: SpeedLimitTagsLayerProps) => {
   const canvas = useRef<HTMLCanvasElement>(null);
   const tooltip = useRef<tooltipInfos | null>();
+  const [images, setImages] = useState<ImagesType>({ questionImage: null, alertFillImage: null });
+
+  useEffect(() => {
+    if (images.questionImage !== null && images.alertFillImage !== null) return;
+    const fetchImages = async () => {
+      const { questionImage, alertFillImage } = await loadSpeedLimitTagsImages();
+      setImages({ questionImage, alertFillImage });
```

Comment:
Nit: no need to destructure here, we can store the whole object in a variable (`const images = await …`)